### ISSUE TITLE
Convert residue likelihoods into probabilities?

### ISSUE THREAD

**Issue by tsjain** [oxpig/AbLang#3]
How does one convert the residue likelihoods into probabilities? Is it by softmax operation?

**Comment by TobiasHeOl** [comment]
Hello! You are correct! You just need to do a softmax across the inner most dimension. I have added the following code snippet as an example for how to do it.

`
import ablang

sequences = ['ASDA']

model = ablang.pretrained()
tokens = model.tokenizer(sequences, pad=True, device=model.used_device)
with torch.no_grad():        
    logits = model.AbLang(tokens)[:,
                                  1:-1, # Removes start and stop token
                                 ] 
    
probabilities = F.softmax(logits,-1)
`

For each residue you will get 24 values. To figure out which value corresponds to which character, you can use the following dictionary (the token == the index).

`
print(model.tokenizer.vocab_to_aa)
`
I hope this helps you!

Best wishes,
Tobias

**Comment by tsjain** [comment]
Thanks. I understand it now.

For the pretrained heavy and light chain models, the mode likelihood automatically trims the output to only present logits for the 20 amino acids. Is that right?

**Comment by TobiasHeOl** [comment]
Hi tsjain, I'm glad it makes more sense now! You are also correct, that the likelihood mode only returns the logits for the 20 amino acids.
